### ISSUE TITLE
v0.2 stuck

### ISSUE THREAD

**Issue by TwistedModding** [focustense/easymod#15]
The latest version is getting stuck on "Done loading headparts, Building NPC Index"

Log Name:      Application
Source:        Application Hang
Date:          6/20/2021 9:23:41 PM
Event ID:      1002
Task Category: (101)
Level:         Error
Keywords:      Classic
User:          N/A
Computer:      DESKTOP-RNBH10N
Description:
The program EasyNPC.exe version 0.2.0.0 stopped interacting with Windows and was closed. To see if more information about the problem is available, check the problem history in the Security and Maintenance control panel.
 Process ID: 6004
 Start Time: 01d7664459db922a
 Termination Time: 2
 Application Path: D:\Modlist\tools\EasyNPC\EasyNPC.exe
 Report Id: cb077145-36b1-4f59-9a25-6ffe7827781a
 Faulting package full name: 
 Faulting package-relative application ID: 
 Hang type: Unknown

Event Xml:
<Event xmlns="http://schemas.microsoft.com/win/2004/08/events/event">
  <System>
    <Provider Name="Application Hang" />
    <EventID Qualifiers="0">1002</EventID>
    <Version>0</Version>
    <Level>2</Level>
    <Task>101</Task>
    <Opcode>0</Opcode>
    <Keywords>0x80000000000000</Keywords>
    <TimeCreated SystemTime="2021-06-21T02:23:41.8266661Z" />
    <EventRecordID>2406</EventRecordID>
    <Correlation />
    <Execution ProcessID="0" ThreadID="0" />
    <Channel>Application</Channel>
    <Computer>DESKTOP-RNBH10N</Computer>
    <Security />
  </System>
  <EventData>
    <Data>EasyNPC.exe</Data>
    <Data>0.2.0.0</Data>
    <Data>6004</Data>
    <Data>01d7664459db922a</Data>
    <Data>2</Data>
    <Data>D:\Modlist\tools\EasyNPC\EasyNPC.exe</Data>
    <Data>cb077145-36b1-4f59-9a25-6ffe7827781a</Data>
    <Data>
    </Data>
    <Data>
    </Data>
    <Data>Unknown</Data>
    <Binary>55006E006B006E006F0077006E0000000000</Binary>
  </EventData>
</Event>

It becomes unresponsive and then I have to close it through task manager. (I waited 10+ minutes to see if it would open, and it wouldnt) reverting to v0.12 allows me to launch the program.

**Comment by focustense** [comment]
Thanks for the report. Since it's not a crash, Windows event log doesn't tell me anything useful. I'd need to see the EasyNPC log file in this case. They're always in %LocalAppData%\EasyNPC. Most likely, the last entry in it will indicate which plugin it's having a problem with.

**Comment by focustense** [comment]
I don't think there's anything in the two additional commits that could result in an infinite loop - at least, nothing that wasn't already there - and that's normally what would be the root cause of an app hang.

One possibility might be that you lost your settings on upgrade (that part is guaranteed) _and_ it's somehow ended up trying to scan your whole hard drive for mods, or something unusual like that. There's a command-line flag I added for my own use which could help if that's what's happening: `/forceintro`. If you put that in the command-line arguments, then it will force the welcome screen, and by extension allow you to change settings before it tries to go through your plugins. Not sure if that's the problem or if correcting it would help, but it's another thing to try.

**Comment by TwistedModding** [comment]
![image](https://user-images.githubusercontent.com/62785432/122788449-f5b4a880-d27b-11eb-9721-94bfe4495a4c.png)
[Log_20210621_103029_0458865.txt](https://github.com/focustense/easymod/files/6687800/Log_20210621_103029_0458865.txt)
 
the /forceintro command let me get to the welcome screen again to put my settings in again, but the app still hangs at the same place indefinetly.

**Comment by TwistedModding** [comment]
![image](https://user-images.githubusercontent.com/62785432/122789077-8c816500-d27c-11eb-857a-fef0f75ab5af.png)
it seems to still be doing something as the cpu usage jumps around a tiny bit. but i left it at this screen for 15 minutes yesterday, and it kept hanging

**Comment by focustense** [comment]
I think I have narrowed it down. Some of the ITPO-detection rules were moved from a UI class into the Rule Set, but the UI model included the master as an explicit "override" while the data model that the Rule Set operates on does not. So there is specifically a problem with ITM records, which I didn't have in my load order when I was testing. I tried it on a much larger load order and was able to repro.

It should be a simple fix since the method is already supposed to handle "no override" cases, just missed this one edge case.
